subctl/issue-89-reconstructing-broker-info.md

```diff
@@ -0,0 +1,71 @@
+# Recover broker-info.subm file from the running Broker
+
+Related Issue:
+[Option for recovering the broker-info.subm file from a running broker cluster](https://github.com/submariner-io/enhancements/issues/143)
+
+## Summary
+
+`broker-info.subm` file is created when a broker is installed. The file contains information critical to have a functional environment.
+There is a possibility that the file could be deleted or lost for one or the other reason. In such cases, no new clusters could be joined.
+
+This proposal describes a way to recover the `broker-info.subm` file from the running broker.
+
+## Proposal
+
+A new `subctl` command will be added that would fetch needed data from the broker and write it to `broker-info.subm` file. The file
+would be Base64 encoded.
+
+## Design details
+
+* subctl command name: `recover-brokerinfo`
```

Comment:
Can you please highlight/show the mandatory flags and the optional flags?

I guess the mandatory flags are `brokerconfig` and all other flags are optional.

Reply:
Neither necessarily has to be mandatory. `--kubeconfig` points to a possible `Submariner` cluster and `--brokerconfig` points to the possible broker cluster. If only `--kubeconfig` is specified and it also has the broker installed then we can proceed w/o needing `--brokerconfig`. Similarly, if only `--brokerconfig` is specified and it also has `Submariner` installed then we can proceed.

Reply:
Right, it was not clear in the description. But anyways, if this is how its implemented, its good.